crates/starknet-devnet-core/src/state/state_readers.rs

```diff
@@ -0,0 +1,121 @@
+        Ok(nonce)
+    }
+
+    fn get_compiled_contract_class(&mut self, class_hash: ClassHash) -> StateResult<ContractClass> {
+        let contract_class = self.class_hash_to_class.get(&class_hash).cloned();
+        match contract_class {
+            Some(contract_class) => Ok(contract_class),
+            _ => Err(StateError::UndeclaredClassHash(class_hash)),
+        }
+    }
+
+    fn get_class_hash_at(&mut self, contract_address: ContractAddress) -> StateResult<ClassHash> {
+        let class_hash =
+            self.address_to_class_hash.get(&contract_address).copied().unwrap_or_default();
+        Ok(class_hash)
+    }
+
+    fn get_compiled_class_hash(
+        &mut self,
+        class_hash: ClassHash,
+    ) -> StateResult<starknet_api::core::CompiledClassHash> {
+        let compiled_class_hash =
+            self.class_hash_to_compiled_class_hash.get(&class_hash).copied().unwrap_or_default();
+        Ok(compiled_class_hash)
+    }
+}
+
+impl State for DictState {
+    fn set_storage_at(
+        &mut self,
+        contract_address: ContractAddress,
+        key: StorageKey,
+        value: StarkFelt,
+    ) -> std::result::Result<(), blockifier::state::errors::StateError> {
+        self.storage_view.insert((contract_address, key), value);
+        Ok(())
+    }
+
+    fn increment_nonce(&mut self, contract_address: ContractAddress) -> StateResult<()> {
+        let current_nonce = self.get_nonce_at(contract_address)?;
+        let current_nonce_as_u64 = usize::try_from(current_nonce.0)? as u64;
+        let next_nonce_val = 1_u64 + current_nonce_as_u64;
+        let next_nonce = Nonce(StarkFelt::from(next_nonce_val));
+        self.address_to_nonce.insert(contract_address, next_nonce);
+
+        Ok(())
+    }
+
+    fn set_class_hash_at(
+        &mut self,
+        contract_address: ContractAddress,
+        class_hash: ClassHash,
+    ) -> StateResult<()> {
+        if contract_address == ContractAddress::default() {
+            return Err(StateError::OutOfRangeContractAddress);
+        }
+
+        self.address_to_class_hash.insert(contract_address, class_hash);
+        Ok(())
+    }
+
+    fn set_contract_class(
+        &mut self,
+        class_hash: ClassHash,
+        contract_class: ContractClass,
+    ) -> StateResult<()> {
+        self.class_hash_to_class.insert(class_hash, contract_class);
+        Ok(())
+    }
+
+    fn set_compiled_class_hash(
+        &mut self,
+        class_hash: ClassHash,
+        compiled_class_hash: CompiledClassHash,
+    ) -> StateResult<()> {
+        self.class_hash_to_compiled_class_hash.insert(class_hash, compiled_class_hash);
+        Ok(())
+    }
+
+    fn to_state_diff(&mut self) -> blockifier::state::cached_state::CommitmentStateDiff {
+        panic!("Should never be called")
+    }
+
+    fn add_visited_pcs(&mut self, _class_hash: ClassHash, _pcs: &std::collections::HashSet<usize>) {
+        todo!("What with this")
```

Comment:
@marioiordanov I was thinking of simply adding a `panic!` here, similar to the method above

Reply:
I think a better one is to be with empty implementation

Reply:
Warning added in https://github.com/0xSpaceShard/starknet-devnet-rs/pull/329/commits/ee01d66cf3ab6fd8d3fab36120c34c7f89af5037

Reply:
Redundant methods removed in commits:
- https://github.com/0xSpaceShard/starknet-devnet-rs/pull/329/commits/6696dc6ef1a78deb34b0ac62a3e1250d926f437f
- https://github.com/0xSpaceShard/starknet-devnet-rs/pull/329/commits/41fa1979abe3489c269b24fd46c705c9d24b6a7b